controllers/imagejob/imagejob_controller.go

```diff
@@ -365,11 +398,34 @@ func (r *Reconciler) handleNewJob(ctx context.Context, imageJob *eraserv1.ImageJ
 			return err
 		}
 		log.Info("Started "+containerName+" pod on node", "nodeName", nodeName)
+
+		// send pod to podQueue to create thread and wait for PodRunning phase
+		waitGroup.Add(1)
+		podQueue <- pod
 	}
 
+	// wait for all PodRunning threads to complete
+	waitGroup.Wait()
```

Comment:
Could end up blocked here? For example, if something went wrong in one of our threads and `waitGroup.Done()` was never called? We may want to have a total timeout here.

Reply:
what would be an appropriate timeout here? each of the threads has a timeout for 5 minutes before waitGroup.Done() from pollImmediate